src/lib/flow/sol-flow-internal.h

```diff
@@ -293,6 +293,8 @@ int sol_flow_builder_add_node_taking_options(
     const struct sol_flow_node_options *options);
 
 #ifdef ENABLE_DYNAMIC_MODULES
+void sol_flow_modules_cache_shutdown(void);
```

Comment:
even if you just use it inside flow, I'd keep all the machinery in the common/core. Then make the old function not-SOL_API and put it inside some private header, expose only the `sol-flow.h` one.

This way, if we need the logic for other parts (likely I'd say), we don't need to move back all of that.